app/vmagent/remotewrite/client.go

```diff
@@ -459,6 +458,35 @@ again:
 		_ = resp.Body.Close()
```

Comment:
It looks like this `resp.Body.Close()` call isn't needed, since it is already called at the line `447` above.